learn/advanced/storage.md

```diff
@@ -71,3 +71,11 @@ These metrics are highly dependent on the machine that is running Meilisearch. R
 It is important to note that **there is no reliable way to predict the final size of a database**. This is true for just about any search engine on the market—we're just the only ones saying it out loud.
 
 Database size is affected by a large number of criteria, including settings, relevancy rules, use of facets, the number of different languages present, and more.
+
+## Soft deletion
+
+Meilisearch renders deleted documents inaccessible to all users but does not immediately remove them from the database. This is a common optimization technique called "soft deletion". Soft deleted documents are permanently deleted at a later date—the precise time depends on index size and available disk space.
+
+When updating a document, Meilisearch deletes the current record and creates a new one containing the new data. This means soft deletion, as well as its associated performance increase, also affect document updates.
+
+If you are developing an application that handles sensitive data, it might be important to check how soft deletion interacts with relevant data retention legislation.
```

Comment:
I am afraid this sentence is afraid the users because every single user thinks they are working on sensitive data 😂 
But I don't know how we can better saying this...
@irevoire maybe?

Reply:
Yeah, I think it's particularly difficult because it's not like we can say "if this is a problem, you can do [something]". At the same time, I think it's a bit dishonest not to mention it? 

Unless my sources are wrong and soft deletion poses no risk for people who need to follow GDPR and etc, in which case I'll delete this straightaway.

Reply:
I know Elastic is doing soft deletion with what they call "segments", maybe we can check on their documentation

Also, you have to know a LOT of programs are doing it also, like gmail. When you delete a mail, even from the trash, it's not immediately deleted from the machine, for optimization.

What we do is not really soft delete, but a temporary soft delete, it's not easy if it's GDPR compliant or not 🫠

Reply:
> What we do is not really soft delete, but a temporary soft delete, it's not easy to know if it's GDPR compliant or not 

Yeah, makes sense. So far, everything I read about soft deletion and GDPR gives me the impression that no one really knows and/or that it might technically violate GDPR, but because the data is very hard to access it's unlikely to become a problem.

I get your point about every user thinking they're handling sensitive data, though. I'll rework it so the information is still there, but it's a bit less scary.